index.js

```diff
@@ -155,6 +155,41 @@ async function fetchChunks(prefix, levelsRemaining, stream) {
     return;
   }
 
+  // Async fetch the last level
+  if (process.env.QUICK_MODE && levelsRemaining == 1) {
+    let promises = [];
+    for (let i = 0; i < 256; i++) {
+      promises.push(fetchChunks(prefix + i.toString(16).padStart(2, "0"), levelsRemaining - 1, stream));
+    }
+    await Promise.all(promises);
+  } else {
+    for (let i = 0; i < 256; i++) {
+      await fetchChunks(prefix + i.toString(16).padStart(2, "0"), levelsRemaining - 1, stream);
+    }
+  }
+}
+
+async function fetchChunksAdvance(prefix, levelsRemaining, stream, exclude_prefix) {
+  if (levelsRemaining <= 0) {
+    if (exclude_prefix.startsWith(prefix)) {
+      if (exclude_prefix == prefix) {
+        return;
+      } else {
+        // exclude_prefix is posterity of prefix, continue searching.
+        levelsRemaining += 1;
+      }
+    } else {
+      const pairs = await provider.send('state_getPairs', [prefix]);
+      if (pairs.length > 0) {
+        separator ? stream.write(",") : separator = true;
+        stream.write(JSON.stringify(pairs).slice(1, -1));
+      }
+      progressBar.update(++chunksFetched);
+      return;
+    }
+
+  }
+
   // Async fetch the last level
   if (process.env.QUICK_MODE && levelsRemaining == 1) {
```

Comment:
It looks like we fetch the state twice.

Reply:
QUICK_MODE | If set, it parallelizes the data download from the RPC endpoint | `NULL` |